src/congestion/BufferOccupancy.cc

```diff
@@ -237,64 +240,27 @@ void BufferOccupancy::performDecrementWindow(u32 _vcIdx) {
   windows_.at(_vcIdx)--;
 }
 
-f64 BufferOccupancy::vcStatusNorm(u32 _outputPort, u32 _outputVc) const {
+f64 BufferOccupancy::vcStatus(
+    u32 _outputPort, u32 _outputVc, bool _normalize) const {
   // return this VC's status
   u32 vcIdx = device_->vcIndex(_outputPort, _outputVc);
-  f64 status;
-  if (!phantom_) {
-    status = ((f64)creditMaximums_.at(vcIdx) - (f64)creditCounts_.at(vcIdx)) /
-             (f64)creditMaximums_.at(vcIdx);
-  } else {
-    status = (((f64)creditMaximums_.at(vcIdx) - (f64)creditCounts_.at(vcIdx) -
-               (f64)windows_.at(vcIdx) * valueCoeff_) /
-              (f64)creditMaximums_.at(vcIdx));
+  f64 status = outstandingFlits_.at(vcIdx);
+  if (phantom_) {
+    status = std::max(0.0, status - (windows_.at(vcIdx) * valueCoeff_));
   }
-  return std::min(1.0, std::max(0.0, status));
-}
-
-f64 BufferOccupancy::vcStatusAbs(u32 _outputPort, u32 _outputVc) const {
-  // return this VC's status in absolute format
-  u32 vcIdx = device_->vcIndex(_outputPort, _outputVc);
-  f64 statusAbs;
-  if (!phantom_) {
-    statusAbs = (f64)flitsOutstanding_.at(vcIdx);
-  } else {
-    statusAbs = (f64)flitsOutstanding_.at(vcIdx) -
-                ((f64)windows_.at(vcIdx) * valueCoeff_);
+  if (_normalize) {
+    status /= normalizationDivisors_.at(vcIdx);
```

Comment:
should there be a check for 0 here (in case initCredits is not called)?

Reply:
Added assert